rclcpp/src/rclcpp/executor.cpp

```diff
@@ -395,6 +395,69 @@ Executor::execute_any_executable(AnyExecutable & any_exec)
   any_exec.callback_group->can_be_taken_from().store(true);
 }
 
+template<typename Function>
+void execute_guarded(
+  const Function & fun,
```

Comment:
nitpick: use full variable names

```suggestion
  const Function & function,
```